platform/linux-generic/include/odp/api/plat/packet_inlines.h

```diff
@@ -216,18 +216,16 @@ _ODP_INLINE uint32_t odp_packet_user_area_size(odp_packet_t pkt)
 
 _ODP_INLINE int odp_packet_user_flag(odp_packet_t pkt)
 {
-	_odp_packet_flags_t flags;
-
-	flags.all_flags = _odp_pkt_get(pkt, uint32_t, flags);
-
-	return flags.user_flag;
+	return _odp_event_hdr_field((odp_event_t)(uintptr_t)pkt, int8_t, user_flag);
 }
 
 _ODP_INLINE void odp_packet_user_flag_set(odp_packet_t pkt, int val)
 {
-	_odp_packet_flags_t *flags = _odp_pkt_get_ptr(pkt, _odp_packet_flags_t, flags);
+	odp_event_t event = (odp_event_t)(uintptr_t)pkt;
+	int8_t *user_flag = _odp_event_hdr_ptr(event, int8_t, user_flag);
 
-	flags->user_flag = !!val;
+	if (odp_likely(*user_flag != -1))
```

Comment:
Is the `if` necessary or leftover from `odp_event_user_flag_set()`? Same comment for event and packet vectors.